pkg/tls/renewer.go

```diff
@@ -20,7 +20,9 @@ const (
 	CAValidityDuration = 365 * 24 * time.Hour
 	// TLSValidityDuration is the valid duration for TLS certificates
 	TLSValidityDuration = 150 * 24 * time.Hour
-	rootCAKey           = "rootCA.crt"
+	// RenewBefore is the renewal time before expiration
+	RenewBefore = 15 * 24 * time.Hour
```

Comment:
sorry for randomly jumping in here, i was digging up on this issue too and found this 😄 

Can we please make the RenewBefore configurable through a CLI flag? We'd like to renew certificates within our own company policies. Though 15d is a good and sane default 👍

Reply:
We'd love to add that. @MariamFahmy98 - wdyt?

Reply:
Done.